src/formatter/ExpressionFormatter.ts

```diff
@@ -381,9 +381,9 @@ export default class ExpressionFormatter {
         case 'preserve':
           return equalizeWhitespace(token.text);
         case 'upper':
-          return token.value;
+          return equalizeWhitespace(token.value);
         case 'lower':
-          return token.value.toLowerCase();
+          return equalizeWhitespace(token.value.toLowerCase());
```

Comment:
The real problem is that `token.value` should already be in upper case and with single spaces (i.e. the `value` field should be in a canonical form).

This allows us to perform simple comparisons like `if (token.value === 'INSERT INTO`)` instead of having to use a regex to compare. So this issue really needs to be handled inside Tokenizer.